paasta_tools/cli/cmds/spark_run.py

```diff
@@ -1028,6 +1031,18 @@ def paasta_spark_run(args):
         )
         return 1
 
+    # validate pool
+    if args.pool:
+        valid_pools = system_paasta_config.get_cluster_pools().get(args.cluster, [])
+        if valid_pools and args.pool not in valid_pools:
```

Comment:
We should have an else condition when `valid_pools` is empty, with a warning message. 
That will ensure us to catch the scenario on why we are not able to fetch `allowed_pools`.

Reply:
Added warning